lib/actions-oclif/local/configure.ts

```diff
@@ -0,0 +1,328 @@
+/**
+ * @license
+ * Copyright 2016-2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { flags } from '@oclif/command';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { stripIndent } from '../../utils/lazy';
+
+interface FlagsDef {
+	help: void;
+}
+
+interface ArgsDef {
+	target: string;
+}
+
+export default class LocalConfigureCmd extends Command {
+	public static description = stripIndent`
+		(Re)configure a balenaOS drive or image.
+
+		Configure or reconfigure a balenaOS drive or image.
+	`;
+
+	public static examples = [
+		'$ balena local configure /dev/sdc',
+		'$ balena local configure path/to/image.img',
+	];
+
+	public static args = [
+		{
+			name: 'target',
+			description: 'path of drive or image to configure',
+			required: true,
+		},
+	];
+
+	public static usage = 'local configure <target>';
+
+	public static flags: flags.Input<FlagsDef> = {
+		help: cf.help,
+	};
+
+	public static root = true;
+
+	public async run() {
+		const { args: params } = this.parse<FlagsDef, ArgsDef>(LocalConfigureCmd);
+
+		const Bluebird = require('bluebird');
+		const path = require('path');
+		const umount = require('umount');
+		const umountAsync = Bluebird.promisify(umount.umount);
+		const isMountedAsync = Bluebird.promisify(umount.isMounted);
+		const reconfix = require('reconfix');
+		const denymount = Bluebird.promisify(require('denymount'));
+		const Logger = await import('../../utils/logger');
+
+		const logger = Logger.getLogger();
+
+		const configurationSchema = await this.prepareConnectionFile(params.target);
+
+		if (await isMountedAsync(params.target)) {
+			await umountAsync(params.target);
+		}
+
+		const dmOpts: any = {};
+		if (process.pkg) {
+			// when running in a standalone pkg install, the 'denymount'
+			// executable is placed on the same folder as process.execPath
+			dmOpts.executablePath = path.join(
+				path.dirname(process.execPath),
+				'denymount',
+			);
+		}
+
+		const dmHandler = (cb: () => void) =>
+			reconfix
+				.readConfiguration(configurationSchema, params.target)
+				.tap((config: any) => {
+					logger.logDebug('Current config:');
+					logger.logDebug(JSON.stringify(config));
+				})
+				.then(this.getConfiguration)
```

Comment:
I would think that these are functionally equivalent (getConfiguration is itself an arrow function), but happy to add (for sure it's clearer).

Reply:
Ah, you're right, just tested accessing `this` inside `getConfiguration`: it works because it was declared as an arrow function. 👍   If it was declared as a regular function, then it is not possible to access `this` (as a reference to the object) inside `getConfiguration` with the "bare" form `.then(this.getConfiguration)`.